### DISCUSSION TITLE
Support for Auhtelia

### DISCUSSION THREAD

**Discussion by dronnikovigor** [Radiokot/photoprism-android-client#115] [Ideas]
Hi!

Do you think it's possible to add support for Authelia authentication ?

Thanks

**Comment by Radiokot** [comment]
Hello Igor. Could you please explain the use-case? I don't understand how Authelia is connected to PhotoPrism.

**Reply by dronnikovigor** [reply]
Authelia is used as an additional security level with ability to enable 2FA. Also it provides SSO for all other services I run on my domain.
So, e.g. 
1. I open https://photoprism.mydomain.net, 
2. I'm being redirected to https://authelia.mydomain.net, 
3. where I login w/ 2FA 
4. and then I'm redirected back to https://photoprism.mydomain.net, where I may login in photoprism. 

If now i'll try to reach https://service2.mydomain.net, authelia will check my session and allow me directly to sevice2.

In comparison to basic auth - Authelia is more convenient and secure. Also it allows to avoid some issues on iPhone.

**Reply by Radiokot** [reply]
Thanks, I got it. 
It seems like in order to support such auth, as well as other solutions like Cloudflare Access or Umbrel, the app needs to:
1. Detect if something is blocking API access and returns HTML when JSON is expected;
2. Open the requested URL in a web view to let the user interact with it;
3. Catch redirect to the PhotoPrism web client;
4. Grab the auth server cookies from the web view and use them to access PhotoPrism API once again;
5. Gracefully log out the user when the auth server session is expired.

I'll consider adding this to the app. Meanwhile, I suggest setting up a bypass to `/api/v1/*` protected with HTTPS and basic auth – then you could enter https://username:password@photoprism.mydomain.net as a library URL.

**Reply by Radiokot** [reply]
Hey Igor, can you test the debug build?
https://vault.bitwarden.com/#/send/xNxOq7cNgUeJt7DyAM4T_Q/NwHUewrsmm50Q93ZM7m-Lg
This will be installed as a separate app, no need to uninstall the one you already have.
I've tested connection through Umbrel and Cloudflare Access auth pages, I think it will work with Authelia as well.

**Reply by dronnikovigor** [reply]
Hi!

Yep, I was able to login and see photos. Checking other stuff

**Reply by Radiokot** [reply]
What do you think should happen once the auth cookies are expired? Logout? Re-appear of the auth web page? Maybe something else?

**Reply by dronnikovigor** [reply]
Hm, i think it will be good to open auth web page, and to save current connection settings.

**Comment by Radiokot** [comment]
[The recent release](https://github.com/Radiokot/photoprism-android-client/releases/tag/1.23.0(36)) has web portal auth support. Thank you for the suggestion, I believe this will be useful for many.
